src/javascript/app/Stores/Modules/Portfolio/portfolio-store.js

```diff
@@ -74,6 +74,9 @@ export default class PortfolioStore extends BaseStore {
             WS.subscribeProposalOpenContract(contract_id.toString(), this.proposalOpenContractHandler, false);
         } else if (act === 'sell') {
             const i = this.getPositionIndexById(contract_id);
+
+            if (i === -1) { return; }
```

Comment:
can you add a comment explaining the race condition? i.e. if contract is already expired in portfolio it won't be added to positions drawer